fs/procfs/fs_procfs.c

```diff
@@ -627,12 +636,18 @@ static int procfs_opendir(FAR struct inode *mountpt, FAR const char *relpath,
 
   if (!relpath || relpath[0] == '\0')
     {
+      size_t num = 0;
+
       /* The path refers to the top level directory.  Allocate the level0
        * dirent structure.
        */
 
+#ifndef CONFIG_FS_PROCFS_EXCLUDE_PROCESS
+      nxsched_foreach(procfs_thread_number, &num);
+#endif
+
       level0 = (FAR struct procfs_level0_s *)
-         kmm_zalloc(sizeof(struct procfs_level0_s));
+         kmm_zalloc(sizeof(struct procfs_level0_s) + sizeof(pid_t) * num) ;
```

Comment:
```suggestion
         kmm_zalloc(sizeof(struct procfs_level0_s) + sizeof(pid_t) * g_npidhash) ;
```

Reply:
g_npidhash may large than the real number of thread

Reply:
Yes, but It is fine iflarger than the real task. line 650 will foreach task again.

Reply:
Ok

Reply:
Done